### PR TITLE
Feature/simplify range queries

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
**Summary of added content:**

- TRACE-Q simplify method that finds the best simplification for a given threshold.
- Range queries for query error

**Relevant files for review:**

- `src/querying/Range_Query.hpp`
- `src/simp-algorithms/TRACE_Q.hpp`
- `src/simp-algorithms/TRACE_Q.cpp`

### CODE REVIEW

File: src/simp-algorithms/TRACE_Q.cpp [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+        // Defines how many time intervals there should be made for each point in the grid.
+        auto time_points = static_cast<int>(std::ceil(1 / time_interval_multiplier));
+
+        for (int x_point = 0; x_point <= points_on_axis; ++x_point) {
+            // Scale the coordinate based on the grid density, the mbr and the current point on the x-axis.
+            auto x_coord = mbr.x_low + x_point * grid_density * (mbr.x_high - mbr.x_low);
+            for (int y_point = 0; y_point <= points_on_axis; ++y_point) {
+                auto y_coord = mbr.y_low + y_point * grid_density * (mbr.y_high - mbr.y_low);
+                for (int t_point = 0; t_point <= time_points; ++t_point) {
+                    auto t = mbr.t_low + t_point * time_interval_multiplier * (mbr.t_high - mbr.t_low);
```
Comment (Kristoffermg) [inline src/simp-algorithms/TRACE_Q.cpp]: Burde den ikke hedde t_coord?

File: src/simp-algorithms/TRACE_Q.cpp [hunk 2]
```diff
@@ -49,13 +70,96 @@ namespace trace_q {
...
+        return mbr;
+    }
+
+    std::vector<std::shared_ptr<spatial_queries::Range_Query>> TRACE_Q::range_query_initialization(
+            data_structures::Trajectory const& trajectory, double x, double y, long double t, MBR const& mbr) const {
+        std::vector<std::shared_ptr<spatial_queries::Range_Query>> result{};
+        std::vector<std::future<spatial_queries::Range_Query>> futures{};
+
+        for (int window_number = 0; window_number < windows_per_grid_point; ++window_number) {
+            auto w_x_low = x - 0.5 * (pow(window_expansion_rate, window_number)
```
Comment (Kristoffermg) [inline src/simp-algorithms/TRACE_Q.cpp]: Jeg synes at de lige så godt kunne hedde window_x_low osv

**Review by Kristoffermg — CHANGES_REQUESTED** [review]

File: src/simp-algorithms/TRACE_Q.cpp (hunk 1 above)
Comment (Sarmisuper) [reply src/simp-algorithms/TRACE_Q.cpp]: renamed til t_interval

File: src/simp-algorithms/TRACE_Q.cpp (hunk 2 above)
Comment (Sarmisuper) [reply src/simp-algorithms/TRACE_Q.cpp]: Donesies

Comment (Kristoffermg) [reply src/simp-algorithms/TRACE_Q.cpp]: thumbs_up

**Review by Kristoffermg — APPROVED** [review]

**Review by Schwartz123 — COMMENTED** [review]
seems so

**Review by Schwartz123 — APPROVED** [review]
looks good

File: src/querying/Range_Query_Test.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#include <algorithm>
+#include "Range_Query_Test.hpp"
+
+namespace spatial_queries {
+        bool Range_Query_Test::operator()(data_structures::Trajectory const& trajectory) {
```
Comment (Kristoffermg) [inline src/querying/Range_Query_Test.cpp]: indentation

**Review by Kristoffermg — CHANGES_REQUESTED** [review]

File: src/querying/Range_Query_Test.cpp (hunk 3 above)
Comment (Sarmisuper) [reply src/querying/Range_Query_Test.cpp]: doneskibidi

**Review by Kristoffermg — APPROVED** [review]

**Pull request merged**